### PR TITLE
Cleaning up public API

### PR DESCRIPTION
**Pull request by lostmsu**
This handles https://github.com/pythonnet/pythonnet/issues/1539

Deleted:
- `PyLong`
- `PyAnsiString`
- a few internal overloads to convert between Python integer type and .NET int types

### Checklist

Check all those that are applicable and complete.

-   [ ] Make sure to include one or more tests for your change
-   [ ] If an enhancement PR, please create docs and at best an example
-   [ ] Updated the [`CHANGELOG`](../blob/master/CHANGELOG.md)

### CODE REVIEW

File: src/embed_tests/TestRuntime.cs [hunk 1]
```diff
@@ -91,6 +97,8 @@ public static void PyCheck_Iter_PyObject_IsIterable_ThreadingLock_Test()
         {
             Runtime.Runtime.Py_Initialize();
 
+            Runtime.Native.ABI.Initialize(Runtime.Runtime.PyVersion);
```
Comment (filmor) [inline src/embed_tests/TestRuntime.cs:100]: What is this for?

Comment (lostmsu) [reply src/embed_tests/TestRuntime.cs:100]: `PyIter_Check` down in the function is not something we import from Python DLL. In Python it is a macro, that we reimplemented in C#. That implementation depends on `TypeOffset` to be initialized, which is handled by `ABI.Initialize`

**Comment by lostmsu** [comment]
@filmor any other comments?

File: src/runtime/delegatemanager.cs [hunk 2]
```diff
@@ -76,7 +76,7 @@ private Type GetDispatcher(Type dtype)
             var cc = CallingConventions.Standard;
             Type[] args = { ptrtype, typetype };
             ConstructorBuilder cb = tb.DefineConstructor(ma, cc, args);
-            ConstructorInfo ci = basetype.GetConstructor(args);
+            ConstructorInfo ci = basetype.GetConstructor(BindingFlags.Instance | BindingFlags.NonPublic, null, args, null);
```
Comment (filmor) [inline src/runtime/delegatemanager.cs:79]: Does this change behaviour?

Comment (lostmsu) [reply src/runtime/delegatemanager.cs:79]: Users can no longer construct their own instances of `Dispatcher` manually (which is the goal of the change anyway).

I doubt anyone did that anyway, it is very obscure.

**Review by filmor — APPROVED** [review]

**Pull request merged**